src/applications/ivc-champva/10-10D/pages/ApplicantSponsorMarriageDetailsPage.jsx

```diff
@@ -82,3 +93,100 @@ export function ApplicantSponsorMarriageDetailsReviewPage(props) {
+        ...titleUI('Applicant date of marriage to sponsor'),
+        dateOfMarriageToSponsor: currentOrPastDateUI({
+          title: 'Date of marriage',
+          errorMessages: {
+            pattern: 'Please provide a valid date',
+            required: 'Please provide the date of marriage',
+          },
+        }),
+        dateOfSeparationFromSponsor: currentOrPastDateUI({
+          title: 'Date of legal separation of the marriage',
+          errorMessages: {
+            pattern: 'Please provide a valid date',
+            required: 'Please provide the date of separation',
+          },
+        }),
+        dateOfMarriageToOtherSpouse: currentOrPastDateUI({
+          title: 'Date of remarriage',
+          errorMessages: {
+            pattern: 'Please provide a valid date',
+            required: 'Please provide the date of remarriage',
+          },
+        }),
+        dateOfSeparationFromOtherSpouse: currentOrPastDateUI({
+          title: 'Date of legal separation of the remarriage',
+          errorMessages: {
+            pattern: 'Please provide a valid date',
+            required: 'Please provide the date of separation',
+          },
+        }),
+      },
+    },
+  },
+  noRemarriageSchema: applicantListSchema(['dateOfMarriageToSponsor'], {
+    titleSchema,
+    dateOfMarriageToSponsor: currentOrPastDateSchema,
+  }),
+  separatedSchema: applicantListSchema(
+    ['dateOfMarriageToSponsor', 'dateOfSeparationFromSponsor'],
+    {
+      titleSchema,
+      dateOfMarriageToSponsor: currentOrPastDateSchema,
+      dateOfSeparationFromSponsor: currentOrPastDateSchema,
+    },
+  ),
+  remarriageSchema: applicantListSchema(
+    ['dateOfMarriageToSponsor', 'dateOfMarriageToOtherSpouse'],
+    {
+      titleSchema,
+      dateOfMarriageToSponsor: currentOrPastDateSchema,
+      dateOfMarriageToOtherSpouse: currentOrPastDateSchema,
+    },
+  ),
+  remarriageSeparatedSchema: applicantListSchema(
+    [
+      'dateOfMarriageToSponsor',
+      'dateOfMarriageToOtherSpouse',
+      'dateOfSeparationFromOtherSpouse',
+    ],
+    {
+      titleSchema,
+      dateOfMarriageToSponsor: currentOrPastDateSchema,
+      dateOfMarriageToOtherSpouse: currentOrPastDateSchema,
+      dateOfSeparationFromOtherSpouse: currentOrPastDateSchema,
+    },
+  ),
+};
+
+// Schemas for follow-up marriage questions
+export const remarriageDetailsSchema = {
+  uiSchema: {
+    applicants: {
+      'ui:options': { viewField: ApplicantField },
+      items: {
+        ...titleUI('Applicant remarriage status'),
+        remarriageIsViable: yesNoUI({
+          title: 'Is the remarriage still viable?',
+          hint: additionalFilesHint,
+          labels: {
+            Y: 'Yes, second marriage is viable',
+            N: 'No, second marriage has legally ended',
+          },
+        }),
+      },
+    },
+  },
+  schema: applicantListSchema(['remarriageIsViable'], {
+    titleSchema,
+    remarriageIsViable: yesNoSchema,
+  }),
+};
```

Comment:
The actual date inputs we collect as well as the "is viable" question for remarriages. These schemas are imported into form.js to build the various conditional pages.